src/scales/scale.time.js

```diff
@@ -224,7 +224,7 @@ function parse(input, scale) {
 	}
 
 	if (options.round) {
-		value = +adapter.startOf(value, options.round);
+		value = adapter.startOf(value, options.round);
```

Comment:
removing + from well behaving adapter calls makes no difference:
https://jsperf.com/plus-in-a-function-call/1

So I would leave these adapter calls intact (its safer to make sure we have number)

Reply:
It seems very specific to the `moment` adapter though. What happens if you're using another date library that doesn't explicitly support the `+` syntax?

Just as importantly, it's also very confusing to me to read the code because it looks like we're returning a non-number type since we're doing a conversion to `number`

Reply:
Its not specific to `moment`. 

An adapter should return a timestamp (aka integer) already.
This `+` here is just in the case when an adapter fails to return timestamp for whatever reason. 

If the adapter does not work correctly, timestamps will go to `0` and that's a good indicator to look at the adapter.

If we don't force adapter results to number in these calls, then we can fail in many other places, where we expect there are only timestamps.

Reply:
When an adapter fails to return a timestamp what is it returning? If it's returning an object from the date library, that's the case where I say it's specific to moment.

I'm not sure that it will always go to `0`. However, that seems worse to me. I would rather the code error out if an adapter is returning unexpected results than to silently corrupt the chart data.

Reply:
> What happens if you're using another date library that doesn't explicitly support the + syntax?

It should never happen, adapters must return a number so if that's not the case, it's a bug.

> ... it's also very confusing to me to read the code ...

To me it makes the code clearer because it states that we are expecting number and that we are not working with any date/time lib custom type.

> its going going to NaN actually, if the value is not convertible to number

That's actually better, that means the error will raise faster.

> However, that seems worse to me. I would rather the code error out if an adapter is returning unexpected results

It's what `+value` is trying to do since `NaN` will generate more visible errors. It also makes the whole code more robust and maintainable because we make sure that the code consuming the adapter results will manipulate numbers only.

For example:

```javascript
ticks.push(time);      // ticks could be any types

// vs 

ticks.push(+time);    // ticks are numbers or NaN
```

I'm +1 with @kurkle

Reply:
> It should never happen, adapters must return a number so if that's not the case, it's a bug.

I'm not sure I understand. If they return a number then `+` does nothing. And if they don't then we don't know what `+` will do

Reply:
>... And if they don't then we don't know what `+` will do

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Arithmetic_Operators#Unary_plus_()

Reply:
Updated. I misunderstood what exactly `+` does